### ISSUE TITLE
Document process to set a WSRequest's body from an array of bytes

### ISSUE THREAD

**Issue by Derya** [playframework/playframework#8396]
I feel that it's not clear from the documentation for JavaWS usage [here](https://www.playframework.com/documentation/2.6.x/JavaWS) how to accomplish this task. Suppose I just have a `byte[]` and I want to set that as the body for my WSRequest. There are different ways to do this but I don't think either of the two I've discovered thus far are clear from the docs. For example, I can use the deprecated method to set the WSRequest body from an InputStream and create a ByteArrayInputStream from the bytes, if I know a bit about InputStreams. Or I could turn the byte array into a ByteString and use InMemoryBodyWritable.

**Comment by wsargent** [comment]
> r I could turn the byte array into a ByteString and use InMemoryBodyWritable.

This is how you would do it.  Technically it would be https://www.playframework.com/documentation/2.6.x/JavaWS#Submitting-Streaming-data

Does the readme in play-ws help? https://github.com/playframework/play-ws#java

**Comment by Derya** [comment]
I've been confused by that line (in the JavaWS page) since earlier today when I read it. What is that `body` function?

And, yes, the readme in play-ws covers this nicely.

**Comment by wsargent** [comment]
So in 

```
CompletionStage<WSResponse> wsResponse = ws.url(url).setBody(body(largeImage)).execute("PUT");
```

this is predicated on being used inside a class that is using [`WSBodyWritables`](https://www.playframework.com/documentation/2.6.x/api/java/play/libs/ws/WSBodyWritables.html)  which supply body:

```
import javax.inject.Inject;

import play.mvc.*;
import play.libs.ws.*;
import java.util.concurrent.CompletionStage;

public class MyClient implements WSBodyReadables, WSBodyWritables {
    private final WSClient ws;

    @Inject
    public MyClient(WSClient ws) {
        this.ws = ws;
    }
    // ...
}
```

* https://www.playframework.com/documentation/2.6.x/JavaWS#Making-a-Request 
* https://www.playframework.com/documentation/2.6.x/JavaWS#Custom-BodyReadables-and-BodyWritables

**Comment by Derya** [comment]
Ah I see, so that doc actually did cover exactly what I wanted. Thanks for the help!

**Comment by Derya** [comment]
Yeah I agree, it's included further up in the code snippet, but the context of that code snippet appears to be just "here what you need to import and what you need to inject to use WSClient", and so I would expect many others to do as I had and not think much about the `implements WSBodyReadables, WSBodyWritables` part. I think there should be a subsection about what WSBodyWritables and WSBodyReadables are.
